stdlib/lists.ncl

```diff
@@ -46,5 +46,33 @@
     any : forall a. (a -> Bool) -> List a -> Bool =
       fun pred l =>
         fold (fun x acc => if pred x then true else acc) l false;
+
+    elem : Dyn -> List -> Bool =
+      fun elt => any (fun x => x == elt);
+
+    partition : forall a. (a -> Bool) -> List a -> {right: List a, wrong: List a} =
+      fun pred l =>
+        let aux = fun acc x => if (pred x) then
+          {right = acc.right @ [x]; wrong = acc.wrong}
+        else
+          {right = acc.right; wrong = acc.wrong @ [x]}
+        in
+        foldl aux {right = []; wrong = []} l;
+
+    generate : forall a. (Num -> a) -> Num -> List a =
+      fun gen n =>
+        if n <= 0 then
+          []
+        else
+          generate gen (n-1) @ [gen n];
+
+    sort : forall a. (a -> a -> Bool) -> List a -> List a =
```

Comment:
`a -> a -> Bool` isn't really cool as a type, because there's no way to infer its semantics just by looking at it :/

Afair the language has no support for tagged unions or symbols to make this nicer, but maybe wrapping it into a record like `{ isLower : a -> a -> Bool }` could help (at the cost of being more verbose)

Reply:
Well, we do have simple enums, so what about something like `a -> a -> <StrictlyLesser, GreaterOrEq>`? Also, there will be documentation at some point.

Reply:
Oooh, then yes, that looks much better imho. Or maybe `<StriclyLesser, Equal, StrictlyGreater>` to be more symmetrical − I guess we'll want a standard definition of an ordering function anyways, so we might as well define it right now

Reply:
I unshamedly copied Rust's [`Ordering`](https://doc.rust-lang.org/std/cmp/enum.Ordering.html) enum.

Reply:
There is nothing to be ashamed of when copying Rust.